01projects/sec03CMakeBackground.md

```diff
@@ -0,0 +1,131 @@
+Recent versions of CMake continue to gain support for GPUs (with better support for CUDA) and other modern technologies.
+Furthermore, the most recent version of CMake is extremely easy to install (even without administrative privileges), and given its ability to easily enforce backward compatibility, there is no reason not to install the latest version (which as of Dec 2022 is `3.25.1`).
+
+### Installing CMake
+
+In this course, we shall use the latest available version of CMake.
+The VSCode development container provided for the course already has this latest version installed.
+However, since we do not invoke the very latest features in the initial weeks of the course, we shall configure it to use a slightly older version for broad compatibility with most platforms and shall help with installing the latest version.
+
+CMake can be installed through a variety of methods on virtually every platform (processor architecture and OS).
+Detailed install instructions shall be made available on the Moodle portal for the course.
+
+## CMake overview
+
+As a high level overview, CMake is a build system generator which provides command line tools that read user-provided plaintext files describing the build requirements of the project, and emits a platform-native build system.
+The plaintext files that describe the project's build requirements are written in CMake's custom language and are named `CMakeLists.txt` (the filenames must have exactly identical casing as given here).
+It can be considered as yet another programming/scripting language with its own rubrics for variables, functions, macros and other control structures (conditional statements, loops etc).
+Owing to the need to cater to a wide variety of platforms, the build requirements are written at a rather high-level describing *what* must be built, and describes the relationships between various dependencies in the projects, and excludes low-level minute details of *how* something should be built.
+
+CMake is much more comprehensive than what was described here, and provides its own integrated support for not only generating the build system, but also performing the build with great control over various aspects, as well as facilitating testing, installation of software onto standard locations on the user's machine, packaging the build's artifacts for wider distribution on various platforms, and various other functionality.
+We recognise the cognitive burden of learning yet another full-fledged language just to describe the project's build requirements!
+Hence, we shall strive to introduce the CMake language and tooling piecemeal over the duration of this course.
+
+### Hierarchical project structure
+
+Large scale research projects typically use a hierarchical project structure using modular code patterns for [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns).
```

Comment:
The phrase "modular code patterns" will definitely require some explanation as even I am not 100% sure what you mean by this